### PR TITLE
Light hash generation does not use strings

### PR DESCRIPTION
**Pull request by mvaligursky**
- optimization to avoid allocations
- added basic tests

### CODE REVIEW

File: src/core/hash.js [hunk 1]
```diff
@@ -19,15 +19,16 @@ function hashCode(str) {
  * Calculates simple hash value of an array of numbers. Designed for performance, but provides good
  * distribution with small number of collisions.
  *.
- * @param {Uint32Array} typedArray - Array of numbers to hash.
+ * @param {number[]|Uint32Array} typedArray - Array of numbers to hash.
```
Comment (willeastcott) [inline src/core/hash.js:22]: The parameter name here no longer matches the function signature.

File: src/core/hash.js [hunk 2]
```diff
@@ -19,15 +19,16 @@ function hashCode(str) {
...
  * @returns {number} Hash value.
+ * @ignore
  */
-function fnv1aHashUint32Array(typedArray) {
+function fnv1aHashUint32Array(array) {
```
Comment (willeastcott) [inline src/core/hash.js:26]: If the function can now also take a number array, is it now misnamed?

Comment (mvaligursky) [reply src/core/hash.js:26]: I knew you would not miss it .. renaming it.

**Review by willeastcott — APPROVED** [review]
LGTM! 😄

**Pull request merged**